src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -158,7 +162,69 @@ private static void createNonImpedantBranch(VariableSet variableSet, EquationSys
         DcEquationSystem.createNonImpedantBranch(variableSet, equationSystem, branch, bus1, bus2);
     }
 
-    private static void createBranchEquations(LfNetwork network, VariableSet variableSet, EquationSystem equationSystem) {
+    private static void createBranchActivePowerTargetEquation(LfBranch branch, PhaseControl.ControlledSide controlledSide,
+                                                              EquationSystem equationSystem, EquationTerm p) {
+        PhaseControl phaseControl = branch.getPhaseControl().orElse(null);
+        if (phaseControl != null
+                && phaseControl.getMode() == PhaseControl.Mode.CONTROLLER
+                && phaseControl.getControlledSide() == controlledSide) {
+            if (phaseControl.getUnit() == PhaseControl.Unit.A) {
+                throw new PowsyblException("Phase control in A is not yet supported");
+            }
+            equationSystem.createEquation(branch.getNum(), EquationType.BRANCH_P)
+                    .addTerm(p);
+        }
+    }
+
+    private static void createImpedantBranch(LfBranch branch, LfBus bus1, LfBus bus2, VariableSet variableSet,
+                                             AcEquationSystemCreationParameters creationParameters,
+                                             EquationSystem equationSystem) {
+        EquationTerm p1 = null;
+        EquationTerm q1 = null;
+        EquationTerm p2 = null;
+        EquationTerm q2 = null;
+        if (bus1 != null && bus2 != null) {
+            boolean deriveA1 = creationParameters.isPhaseControl() && branch.getPhaseControl().isPresent();
+            p1 = new ClosedBranchSide1ActiveFlowEquationTerm(branch, bus1, bus2, variableSet, deriveA1, false);
+            q1 = new ClosedBranchSide1ReactiveFlowEquationTerm(branch, bus1, bus2, variableSet, deriveA1, false);
+            p2 = new ClosedBranchSide2ActiveFlowEquationTerm(branch, bus1, bus2, variableSet, deriveA1, false);
+            q2 = new ClosedBranchSide2ReactiveFlowEquationTerm(branch, bus1, bus2, variableSet, deriveA1, false);
+        } else if (bus1 != null) {
+            p1 = new OpenBranchSide2ActiveFlowEquationTerm(branch, bus1, variableSet);
+            q1 = new OpenBranchSide2ReactiveFlowEquationTerm(branch, bus1, variableSet);
+        } else if (bus2 != null) {
+            p2 = new OpenBranchSide1ActiveFlowEquationTerm(branch, bus2, variableSet);
+            q2 = new OpenBranchSide1ReactiveFlowEquationTerm(branch, bus2, variableSet);
+        }
+
+        if (p1 != null) {
+            equationSystem.createEquation(bus1.getNum(), EquationType.BUS_P).addTerm(p1);
+            branch.setP1(p1);
+
+            if (creationParameters.isPhaseControl()) {
+                createBranchActivePowerTargetEquation(branch, PhaseControl.ControlledSide.ONE, equationSystem, p1);
```

Comment:
Maybe we have to put a comment here to say that only local regulation would be supported ?

Reply:
Do we always have side 1 in IIDM = side 1 in the LFBranch ?

Reply:
We should indeed add a TODO to show that this part of the code will have to be updated for remote control.
Yes side 1 in IIDM is always side 1 in LF.

Reply:
I understand that in the code, we can only have a variation on A1 but we can have 2 controller sides: side 1, that means a local control and side 2, that means like a remote control. Are we ok about that ?